frontend/model/state.js

```diff
@@ -22,6 +24,57 @@ import identityGetters from './contracts/shared/getters/identity.js'
 import notificationModule from '~/frontend/model/notifications/vuexModule.js'
 import settingsModule from '~/frontend/model/settings/vuexModule.js'
 import chatroomModule from '~/frontend/model/chatroom/vuexModule.js'
+import { CHELONIA_RESET, CONTRACTS_MODIFIED } from '../../shared/domains/chelonia/events.js'
+
+// Wrapper function for performing contract upgrades and migrations
+// TODO: Consider moving this function into a different file
+const contractUpdate = (initialState: Object, updateFn: (state: Object, contractIDHints: ?string[]) => any, contractType: ?string) => {
+  // Wrapper for the update function. This performs a common check, namely that
+  // the contract is of a certain type, which helps return early
+  const wrappedUpdateFn = contractType
+  // The following disable is because eslint gets confused with 'Object'
+  // eslint-disable-next-line no-use-before-define
+    ? (state: Object, contractIDHints: ?string[]) => {
+        if (Array.isArray(contractIDHints)) {
+          if (!contractIDHints.reduce((acc, contractID) => {
+            return (acc || state.contracts[contractID]?.type === contractType)
+          }, false)) return
+        }
+        updateFn(state, contractIDHints)
+      }
+    : updateFn
+
+  const resetHandler = () => {
+    sbp('okTurtles.events/off', CONTRACTS_MODIFIED, modifiedHandler)
+  }
```

Comment:
Since this is 1 line and it is only used once with a `/once` listener, it can be inlined as an anonymous function below.

Reply:
I thought that you preferred named functions over inlined ones (and it's also more consistent in this case)

Reply:
I do, usually, it especially helps when there's a large or confusing chunk of logic to simplify, or to DRY things. In this case it's simple enough to inline and there's no DRY violation. However, it's not a big to leave as-is too of you prefer.